### PR TITLE
RM-8395 Package res folder based on modern practices in Nuget package

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-8395

### CODE REVIEW

File: Build/NuSpec/Web.targets [hunk 1]
```diff
@@ -9,16 +9,40 @@
   </ItemGroup>
 
   <PropertyGroup>
+    <!-- Old .NET Framwork ASP.NET projects are detected using the project type guid -->
     <WebApplicationProjectTypeGuid>{349c5851-65df-11da-9384-00065b846f21}</WebApplicationProjectTypeGuid>
-    <RemotionWebResource_IsWebApplication>$(ProjectTypeGuids.ToLower().Contains($(WebApplicationProjectTypeGuid)))</RemotionWebResource_IsWebApplication>
+    <_isNetFramworkWebApplicationProject>$(ProjectTypeGuids.ToLower().Contains($(WebApplicationProjectTypeGuid)))</_isNetFramworkWebApplicationProject>
+    <!-- .Contains returns 'True'/'False' so we need to normalize to lowercase here -->
+    <_isNetFramworkWebApplicationProject>$(_isNetFramworkWebApplicationProject.ToLower())</_isNetFramworkWebApplicationProject>
```
Comment (MichaelKetting) [inline Build/NuSpec/Web.targets:16]: Wouldn't it be cleaner to do this all in one?

File: Build/NuSpec/Web.targets [hunk 2]
```diff
@@ -9,16 +9,40 @@
...
+    <_isWebApplicationProject Condition="'$(_isNetFramworkWebApplicationProject)' == 'true' OR '$(_isAspNetCoreWebApplicationProject)' == 'true'">true</_isWebApplicationProject>
+    <!-- Determine if we use symlinks (old) or contentFiles (new). For .NET Framework ASP.NET projects we choose symlinks, otherwise we use contentFiles. -->
+    <_useSymlink>false</_useSymlink>
+    <_useSymlink Condition="'$(_isNetFramworkWebApplicationProject)' == 'true'">true</_useSymlink>
   </PropertyGroup>
 
-  <Target Name="RemotionWebResource_DeleteResFolder" Condition="'$(RemotionWebResource_IsWebApplication)' == 'True'">
+  <PropertyGroup>
+    <RemotionWebResource_IsWebApplication Condition="'$(RemotionWebResource_IsWebApplication)' == ''">$(_isWebApplicationProject)</RemotionWebResource_IsWebApplication>
+    <RemotionWebResource_TargetPath Condition="'$(RemotionWebResource_TargetPath)' == ''">res</RemotionWebResource_TargetPath>
```
Comment (MichaelKetting) [inline Build/NuSpec/Web.targets:28]: Do we have logic duplication with `%(RemotionWebResource_Package.ProjectResFolder)%`?

**Review by MichaelKetting — COMMENTED** [review]
Some questions.

File: Build/NuSpec/Web.targets (hunk 1 above)
Comment (a-ctor) [reply Build/NuSpec/Web.targets:16]: Problem is that I could not get the conversion from `bool` to `string` to work as `.ToLower` only works on string. This way, the conversion is done implicitly.

File: Build/NuSpec/Web.targets (hunk 2 above)
Comment (a-ctor) [reply Build/NuSpec/Web.targets:28]: Not really, the `RemotionWebResource_Package` is only needed for the symlink method and this only works for the contentFiles variant. In the case of contentFiles we do not need the complexity introduced by `RemotionWebResource_Package` as we do not need targets that could be overriden by another project

File: Build/ProjectTypes/WebLibrary.targets [hunk 3]
```diff
@@ -5,6 +5,10 @@
       <Pack>true</Pack>
       <PackagePath>build\$(AssemblyName).targets</PackagePath>
     </Content>
+    <Content Include="$(MSBuildThisFileDirectory)..\NuSpec\Web.targets" CopyToOutputDirectory="Never">
+      <Pack>true</Pack>
+      <PackagePath>buildTransitive\$(AssemblyName).targets</PackagePath>
+    </Content>
```
[the remark this answers is not in the document]
Comment (a-ctor) [reply Build/ProjectTypes/WebLibrary.targets:11]: - `build` works for SDK and non-SDK on the project that references the package
- `buildTransitive` works only for SDK and projects that transitively reference the package

We depend on the file name of the targets file so we couldn't separate the targets. We also have a bit of shared code that would need to be duplicated (as it is unclear which target file would execute first). Then there is the issue that all target files are loaded regardless of if it is SDK or not so we need to do the detection in both files...

I don't think that splitting up the code adds any benefit.

File: Build/NuSpec/Web.targets (hunk 1 above)
Comment (MichaelKetting) [reply Build/NuSpec/Web.targets:16]: Ah, yes. Personally, I probably would go with `.ToStirng().ToLower()` in that case. As you like?

File: Build/ProjectTypes/WebLibrary.targets (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Build/ProjectTypes/WebLibrary.targets:11]: Hmm...it still feels fishy to me but here's to hoping for kicking .net full framework all together in the next major version so this is a temporary issue anyway.

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**